packages/dapp-high-roller/src/components/app-provider/app-provider.tsx

```diff
@@ -0,0 +1,94 @@
+import { Component, Element, Prop, State } from "@stencil/core";
+import { injectHistory, RouterHistory } from "@stencil/router";
+
+import CounterfactualTunnel from "../../data/counterfactual";
+import { AppInstance } from "../../data/mock-app-instance";
+import MockNodeProvider from "../../data/node-provider";
+import { cf, Node } from "../../data/types";
+
+declare var NodeProvider;
+declare var cf;
+
+@Component({
+  tag: "app-provider"
+})
+export class AppProvider {
+  @Element() private el: HTMLStencilElement = {} as HTMLStencilElement;
+  @Prop() history: RouterHistory = {} as RouterHistory;
+  @Prop() updateAppInstance: (appInstance: AppInstance) => void = () => {};
+  @Prop() updateAppFactory: (appFactory: cf.AppFactory) => void = () => {};
+  @Prop({ mutable: true })
+  nodeProvider: MockNodeProvider = {} as MockNodeProvider;
+
+  @Prop({ mutable: true }) cfProvider: cf.Provider = {} as cf.Provider;
+  @Prop({ mutable: true }) appFactory: cf.AppFactory = {} as cf.AppFactory;
+
+  @State() appInstance: AppInstance = {} as AppInstance;
+
+  componentWillLoad() {
+    if (
+      this.history &&
+      this.history.location &&
+      this.history.location.query &&
+      this.history.location.query.standalone
+    ) {
+      // This is supposed to work as per: https://stenciljs.com/docs/router-tutorials#route-query-parameters
+      // However this.history seems to always be undefined... Maybe because this is the root?
+      return;
+    }
+
+    const params = new URLSearchParams(window.location.search);
+
+    // Using promise syntax because lifecycle events aren't
+    // async/await-friendly.
```

Comment:
i thought i saw us using async/await in other hooks?

Reply:
@joelalejandro What do you think?

Reply:
My bad, @Alonski. We should be able to use async/await with no trouble. At the time I didn't understand lifecycle events + tunnels correctly.

Reply:
Ok thanks @joelalejandro
Adding a TODO to fix this